src/Symfony/Component/Yaml/Dumper.php

```diff
@@ -84,6 +84,30 @@ public function dump($input, $inline = 0, $indent = 0, $flags = 0)
             $isAHash = array_keys($input) !== range(0, count($input) - 1);
 
             foreach ($input as $key => $value) {
+                if ($inline - 1 > 0 && Yaml::DUMP_MULTI_LINE_AS_BLOCK & $flags && is_string($value) && false !== strpos($value, "\n")) {
+                    $output .= sprintf(
+                        "%s%s%s |\n",
+                        $prefix,
+                        $isAHash ? Inline::dump($key, $flags).':' : '-',
+                        ''
+                    );
+
+                    foreach (preg_split("/\n|\r\n/", $value) as $row) {
```

Comment:
currently, you are putting newline chars in the string passed as the regex, meaning that the regex is effectively this:

```
/
|
/
```

It would be better to pass ``\n`` to the regex (i.e. not using them as PHP escape sequences but as PCRE ones), which is done by using single quotes for the string (which disables PHP escape sequences)